jbehave-support-core/src/main/java/org/jbehavesupport/core/splunk/SplunkSteps.java

```diff
@@ -0,0 +1,98 @@
+
+    @Given("the Splunk search query is performed:$splunkQuery")
+    @When("the Splunk search query is performed:$splunkQuery")
+    public void executeSplunkSearchQuery(ExpressionEvaluatingParameter<String> splunkQueryExpression) {
+        testContext.put(SPLUNK_RESULT_KEY, splunkClient.search(splunkQueryExpression.getValue()));
+    }
+
+    @Given("the Splunk search query is performed within [$earliestTime] and [$latestTime]:$splunkQuery")
+    @When("the Splunk search query is performed within [$earliestTime] and [$latestTime]:$splunkQuery")
+    public void executeSplunkSearchQuery(ExpressionEvaluatingParameter<String> earliestTimeExpression, ExpressionEvaluatingParameter<String> latestTimeExpression, ExpressionEvaluatingParameter<String> splunkQueryExpression) {
+        testContext.put(SPLUNK_RESULT_KEY, splunkClient.search(splunkQueryExpression.getValue(), earliestTimeExpression.getValue(), latestTimeExpression.getValue()));
+    }
+
+    @Then("the Splunk search result set has $rowCount row(s)")
+    public void compareSplunkSearchResultSetRowCount(ExpressionEvaluatingParameter<Integer> rowCount) {
+        assertThat(getSplunkSearchResult())
+            .as("returned Splunk search result set row count does not match expected count")
+            .hasSize(rowCount.getValue());
+    }
+
+    @Then("the Splunk search result match these rules:$rulesData")
+    public void matchSplunkSearchResultAgainstRules(String verifiersData) {
+        compareExampleTableWithListOfSplunkSearchResultEntries(examplesTableFactory.convertValue(verifiersData, null));
+    }
+
+    private void compareExampleTableWithListOfSplunkSearchResultEntries(ExamplesTable verifiersData) {
+        List<SplunkSearchResultEntry> splunkSearchData = getSplunkSearchResult();
+        notNull(verifiersData, "verifier meta data can't be null");
+        notNull(splunkSearchData, "search result set data can't be null");
+        isTrue(verifiersData.getHeaders().size() == 2 && verifiersData.getHeaders().contains(VERIFIER), "searchData must have one search data column and a verifier)");
+
+        assertThat(splunkSearchData)
+            .as("no Splunk search query match")
+            .isNotEmpty();
+        verifySearchResultEntries(verifiersData, splunkSearchData);
+    }
+
+    private void verifySearchResultEntries(ExamplesTable verifiersData, List<SplunkSearchResultEntry> splunkSearchData) {
+        convertTable(verifiersData).forEach(row ->
+            assertThat(splunkSearchData)
+                .as("all found Splunk search result data meet all verifier definitions")
+                .allSatisfy(
+                    splunkSearchResultEntry -> {
+                        String verifierName = row.get(VERIFIER);
+                        String verifierData = row.get(DATA);
+                        Verifier verifier = verifierResolver.getVerifierByName(verifierName);
+                        if (Objects.nonNull(verifier)) {
```

Comment:
should not be needed - default verifier resolver should throw exception if its not found (altough user can redefine it by its own resolver)

Reply:
Ok, I will remove as it is handled by verifierResolver.getVerifierByName() whose validations I did not check before.